rust/vectordb/src/table.rs

```diff
@@ -113,25 +211,26 @@ impl Table {
                     message: e.to_string(),
                 },
             })?;
-        Ok(Table {
+        Ok(NativeTable {
             name: name.to_string(),
             uri: uri.to_string(),
-            dataset: Arc::new(dataset),
+            dataset: Arc::new(Mutex::new(dataset)),
             store_wrapper: write_store_wrapper,
         })
     }
 
-    /// Checkout a specific version of this [`Table`]
+    /// Make a new clone of the intenral lance dataset.
+    pub(crate) fn clone_inner_dataset(&self) -> Result<Dataset> {
+        Ok(self.dataset.lock()?.clone())
```

Comment:
Why not unwrap/expect here?

Reply:
Good point. sure i can do expect, i guess a poisoned lock is hardly recoverable anyway.